### ISSUE TITLE
Support for PowerShell 7.0

### ISSUE THREAD

**Issue by powershellking** [PSAppDeployToolkit/PSAppDeployToolkit#475]
Yesterday PowerShell 7.0 was released. Are there plans to add support for PowerShell 7.0 in PSAppDeployToolkit

**Comment by luki1412** [comment]
PowerShell 7 is still a "Core" edition. It is not the same like Windows Poweshell. Altough this release should support all features Windows Powershell supports, unlike vs 6, it is based on Net Core not Net Framework. Im pretty sure this will require more than just renaming powershell.exe to pwsh.exe. So we need to create another branch or for the project.

**Comment by jimmylevell** [comment]
Is it planned to support powershell 7.0 in future?

**Comment by eFiggy** [comment]
Not that I want to necro seemingly-dead thread, but is there a plan for PSADT to support PowerShell 7 (7.2.2 is the latest LTS release as of this writing)? Has the team determined what the roadblocks would be?

**Comment by angrycuban13** [comment]
Hey guys, when running `AppDeployToolkitHelp.ps1` from a Powershell 7 console it throws all kinds of **non-terminating** errors because PSAPDT is still using `Get-WMIObject`. Any plans to change this? 

Sorry for necro'ing the thread

**Comment by PHCVergouwe** [comment]
Replacing get-wmiobject with get-ciminstance (and the proper parameters) would be a starting point? :D

**Comment by angrycuban13** [comment]
> Replacing get-wmiobject with get-ciminstance (and the proper parameters) would be a starting point? :D

I understand what the fix is and understand how to do it but that wasn't the point of my post.

**Comment by JonathanPitre** [comment]
Any progress on this one? PWSH 7 support is long overdue!

**Comment by soroshsabz** [comment]
ITNOA

I get below errors in PowerShell 7.3

```
> import-module PSADT
InvalidOperation: C:\Users\sooro\Documents\PowerShell\Modules\PSADT\3.9.3\Public\AppDeployToolkitMain.ps1:325
Line |
 325 |  [String]$envCLRVersion = $envCLRVersion.ToString()
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | You cannot call a method on a null-valued expression.
Add-Type: C:\Users\sooro\Documents\PowerShell\Modules\PSADT\3.9.3\Public\AppDeployToolkitMain.ps1:15732
 Line |
15732 |      Add-Type -Path $appDeployCustomTypesSourceCode -ReferencedAssembl …
      |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      | C:\Users\sooro\Documents\PowerShell\Modules\PSADT\3.9.3\Public\AppDeployToolkitMain.cs(262,17): error CS0246:
      | The type or namespace name 'List<>' could not be found (are you missing a using directive or an assembly
      | reference?)   public static List<IntPtr> EnumWindows()                 ^
InvalidOperation: C:\Users\sooro\Documents\PowerShell\Modules\PSADT\3.9.3\Public\AppDeployToolkitMain.ps1:325
Line |
 325 |  [String]$envCLRVersion = $envCLRVersion.ToString()
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | You cannot call a method on a null-valued expression.
[06-24-2023 01:58:49.935] [PSAppDeployToolkitExt] :: Script [C:\Users\sooro\Documents\PowerShell\Modules\PSADT\3.9.3\Public\AppDeployToolkitExtensions.ps1] dot-source invoked by [C:\Users\sooro\Documents\PowerShell\Modules\PSADT\3.9.3\Public\AppDeployToolkitHelp.ps1]
[06-24-2023 01:58:49.957] [PSAppDeployToolkitHelp] :: Load [App Deploy Toolkit Help] console...
Write-Error: Failed to import function C:\Users\sooro\Documents\PowerShell\Modules\PSADT\3.9.3\Public\AppDeployToolkitMain.ps1: You cannot call a method on a null-valued expression.
```

**Comment by mjr4077au** [comment]
> ITNOA
> 
> I get below errors in PowerShell 7.3
Under this line: https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/blob/18e1bb70ddf7561d32f32a726c18d33c00ea272c/Toolkit/AppDeployToolkit/AppDeployToolkitMain.ps1#L15796, add the following: `if ($PSVersionTable.PSEdition.Equals('Core')) {$ReferencedAssemblies += 'System.Collections', 'System.Text.RegularExpressions', 'System.Security.Principal.Windows', 'System.ComponentModel.Primitives', 'Microsoft.Win32.Primitives'}`

Once that's done, delete everything to do with `$envCLRVersion` (who cares about this, if you're using anything less than PS 5.1 then that's your problem). That'll be enough to get PSADT up and running, for the most part.

**Comment by mjr4077au** [comment]
PowerShell 7.x support, at least initially, should be fixed via #882 and #885.
